### PR TITLE
Sj/6318/handle unauthorized

### PR DESCRIPTION
**Pull request by sago2k8**
Handle unauthorised on console when the token is invalid:

this also simplifies the embedded console since there won't be an user ever, we only rely in the token to handle authz.
## ref: 

[issue](https://github.com/redpanda-data/cloudv2/issues/6318)

Screenshot 2023-06-06 at 09 01 33

### CODE REVIEW

File: frontend/src/components/misc/Error401Page.tsx [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+} from '@redpanda-data/ui';
+import PageContent from './PageContent';
+
+import { api } from '../../state/backendApi';
+
+import { appGlobal } from '../../state/appGlobal';
+import { observer } from 'mobx-react';
+
+const ErrorPage = observer(() => {
+    api.refreshSupportedEndpoints().then(r => {
```
Comment (sago2k8) [inline frontend/src/components/misc/Error401Page.tsx:34]: If the resource is accesible redirect to `/overview`

File: frontend/package.json [hunk 2]
```diff
@@ -26,7 +26,7 @@
         "@heroicons/react": "^1.0.6",
         "@monaco-editor/react": "^4.4",
         "@primer/octicons-react": "^17",
-        "@redpanda-data/ui": "^2.24.0",
+        "@redpanda-data/ui": "^2.40.0",
```
Comment (malinskibeniamin) [inline frontend/package.json:29]: There's a small issue with the sidebar in this version, if you approve my PR to revert it, we can use 2.41

File: frontend/src/components/misc/Error401Page.tsx [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+                                the error persists <Button as={Link} margin={0} padding={0} variant="link"
+                                    isExternal={true}
+                                    href="https://support.redpanda.com/hc/en-us/requests/new">contact
+                                    support</Button>.
+                            </Text>
+                        </Box>
+                    </VStack>
+                    <Hide below="sm">
+                        <Image boxSize="sm" src={Avatars.errorBananaSlipSvg} alt="error-401" />
+                    </Hide>
```
Comment (malinskibeniamin) [inline frontend/src/components/misc/Error401Page.tsx:66]: We already have an error page, which can have customized title and description - can that be used?

Comment (sago2k8) [reply frontend/src/components/misc/Error401Page.tsx:66]: The error page styles are incorrect, plan to fix it in the future, it uses  `100vw` wich is not the case for the embeded console in this case

Comment (malinskibeniamin) [reply frontend/src/components/misc/Error401Page.tsx:68]: Understood, yes correct it was meant to be used as a full error page, hence the 100vw/vh but I see your point, in that case let's use this approach. No worries man

**Review by malinskibeniamin — APPROVED** [review]

File: frontend/src/components/misc/Error401Page.tsx [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+
+import { appGlobal } from '../../state/appGlobal';
+import { observer } from 'mobx-react';
+import fetchWithTimeout from '../../utils/fetchWithTimeout';
+import { config } from '../../config';
+
+const ErrorPage = observer(() => {
+    fetchWithTimeout(`${config.restBasePath}/console/endpoints`, 5 * 1000).then(r => {
+        if (r.ok) {
+            appGlobal.history.replace('/overview');
```
Comment (sago2k8) [inline frontend/src/components/misc/Error401Page.tsx:37]: redirect to overview, then hard refresh if the user is `authorized`

File: frontend/src/EmbeddedApp.tsx [hunk 5]
```diff
@@ -1,12 +1,13 @@
...
+ * Use of this software is governed by the Business Source License
  * included in the file https://github.com/redpanda-data/redpanda/blob/dev/licenses/bsl.md
  *
  * As of the Change Date specified in that file, in accordance with
  * the Business Source License, use of this software will be governed
  * by the Apache License, Version 2.0
- */
-
-import { useEffect } from 'react';
+ */import { useEffect } from 'react';
```
Comment (rikimaru0345) [inline frontend/src/EmbeddedApp.tsx:10]:
```suggestion
 */
 
 import { useEffect } from 'react';
```

line break between import and header

File: frontend/src/components/pages/connect/Connector.Details.tsx [hunk 6]
```diff
@@ -139,7 +139,7 @@ const KafkaConnectorMain = observer(
                 </Tooltip>
             </Flex>
 
-            <Tabs isLazy marginBlock="2"
+            <Tabs marginBlock="2"
```
Comment (rikimaru0345) [inline frontend/src/components/pages/connect/Connector.Details.tsx:142]: why do you remove the `isLazy` setting?

File: frontend/src/components/pages/connect/CreateConnector.tsx [hunk 7]
```diff
@@ -157,8 +157,9 @@ const ConnectorType = observer(
                                 component: <></>
                             },
                         ]} marginBlock="2"
-                            onChange={index => state.tabFilter = tabFilterModes[index]}
-                            index={tabFilterModes.indexOf(state.tabFilter)}
+                            onChange={(_, key) => {
+                                state.tabFilter = key as typeof tabFilterModes[number];
+                            }}
                         />
```
Comment (rikimaru0345) [inline frontend/src/components/pages/connect/CreateConnector.tsx:163]: why change this to remove index?

**Review by rikimaru0345 — CHANGES_REQUESTED** [review]

**Pull request merged**